shell/common/pointer_data_dispatcher.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef POINTER_DATA_DISPATCHER_H_
+#define POINTER_DATA_DISPATCHER_H_
+
+#include "flutter/runtime/runtime_controller.h"
+#include "flutter/shell/common/animator.h"
+
+namespace flutter {
+
+class PointerDataDispatcher {
+ public:
+  virtual void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                              uint64_t trace_flow_id) = 0;
+  virtual void OnRender() = 0;
+  virtual ~PointerDataDispatcher() {}
+};
+
+class DefaultPointerDataDispatcher : public PointerDataDispatcher {
+ public:
+  DefaultPointerDataDispatcher(Animator& animator,
+                               RuntimeController& runtime_controller)
+      : runtime_controller_(runtime_controller), animator_(animator) {}
+
+  void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                      uint64_t trace_flow_id) override;
+  void OnRender() override {}  // Intentional no-op
+
+  virtual ~DefaultPointerDataDispatcher() {}
+
+ protected:
+  RuntimeController& runtime_controller_;
+  Animator& animator_;
+};
+
+class IosPointerDataDispatcher : public DefaultPointerDataDispatcher {
```

Comment:
The documentation you added inline in the implementation of the DispatchPacket can be added here instead in doxygen format to clarify the purpose if this class. Also, maybe a more descriptive name for this class would be better. One that describes what this class does and not where this class is used.

Reply:
Done.